# Experiment 42c74423164779e532b7bcaf48078f34ec66f01abe7d69631f96eaf115ea15e7
# Base 00c84a4a1c32ae81f12092c81759cf818aee2fee
diff --git a/src/tact_tek.cpp b/src/tact_tek.cpp
index 70a1ff06e..0bae3f39f 100644
--- a/src/tact_tek.cpp
+++ b/src/tact_tek.cpp
@@ -175,21 +175,19 @@ void TRIBE_Action_Make_Tech::get_info(short& tech_id_out,
                                        short& progress_out,
                                        char* name_out,
                                        short max_length) {
-    register TRIBE_Action_Make_Tech* self = this;
-    char* destination = name_out;
-
-    tech_id_out = self->tech_id;
-    if (self->state == 1) {
+    tech_id_out = this->tech_id;
+    if (this->state == 1) {
         progress_out = 100;
     } else {
-        TRIBE_Player* player = static_cast<TRIBE_Player*>(self->obj->owner);
-        progress_out = player->tech_tree->get_progress(self->tech_id);
+        progress_out = static_cast<TRIBE_Player*>(this->obj->owner)
+                           ->tech_tree->get_progress(this->tech_id);
     }
 
-    if (destination != 0) {
-        int length = max_length;
-        TRIBE_Player* player = static_cast<TRIBE_Player*>(self->obj->owner);
-        strncpy(destination, player->tech_tree->get_name(self->tech_id), length);
-        destination[length - 1] = 0;
+    if (name_out != 0) {
+        strncpy(name_out,
+                static_cast<TRIBE_Player*>(this->obj->owner)
+                    ->tech_tree->get_name(this->tech_id),
+                max_length);
+        name_out[max_length - 1] = 0;
     }
 }
